src/nav/nav.ts

```diff
@@ -196,6 +214,68 @@ export class NgbNav implements AfterContentInit {
     }
   }
 
+  onKeyDown(event: KeyboardEvent) {
+    if (this.roles !== 'tablist') {
+      return;
+    }
+    // tslint:disable-next-line: deprecation
+    const key = event.which;
+    const tabs: any[] = this.items.filter(x => !x.disabled);
+
+    const navLinkElements = tabs.map(
```

Comment:
I suggest to simplify this significantly using `QueryList<NgbNavLink>`:

1. Need to inject the `public elRef: ElementRef` in the `NgbLavLink` constructor.
2. Query links same as items in the `NgbNav`:

```ts
  @ContentChildren(NgbNavItem) items: QueryList<NgbNavItem>;
  @ContentChildren(forwardRef(() => NgbNavLink), {descendants: true}) links: QueryList<NgbNavLink>;
```

3. Use only enabled links (note that link knows its navItem):

```ts
    const enabledLinks = this.links.filter(link => !link.navItem.disabled);
    const {length} = enabledLinks;

    let position = -1;
    enabledLinks.forEach((link, index) => {
      if (link.elRef.nativeElement === this._document.activeElement) {
        position = index;
      }
    });
```

4. Just do this to focus/select:

```ts
      if (this.selectOnKeyboardEvent) {
        this.select(enabledLinks[position].navItem.id);
      }
      enabledLinks[position].elRef.nativeElement.focus();
```